File: examples/Delphi/Simple/GUIControls/BoldImage/fViewAutoSize.pas
unit fViewAutoSize;

interface

uses
  SysUtils,
  Classes,
  Controls,
  Forms,
  Dialogs,
  BoldImage,
  ComCtrls,
  ToolWin,
  StdCtrls,
  Boldhandles,
  BoldControlsDefs,
  BoldSubscription,
  BoldRootedHandles,
  BoldExpressionHandle,
  BoldReferenceHandle;

type
  TfrmImageViewer = class(TForm)
    ToolBar1: TToolBar;
    cboScale: TComboBox;
    btnScale100: TToolButton;
    ToolButton2: TToolButton;
    ScrollBox1: TScrollBox;
    BoldImage: TBoldImage;
    ToolButton1: TToolButton;
    btnAuto: TToolButton;
    btnScale: TToolButton;
    behImage: TBoldReferenceHandle;
    procedure cboScaleChange(Sender: TObject);
    procedure btnScale100Click(Sender: TObject);
    procedure btnAutoClick(Sender: TObject);
    procedure btnScaleClick(Sender: TObject);
    procedure BoldImageResize(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$R *.DFM}

procedure TfrmImageViewer.cboScaleChange(Sender: TObject);
begin
  btnScale.Click;
  btnScale.Down := True;
  BoldImage.Scale := StrToInt(cboScale.Text);
end;

procedure TfrmImageViewer.btnScale100Click(Sender: TObject);
begin
  btnScale.Click;
  btnScale.Down := True;
  BoldImage.Scale := 100;
end;

procedure TfrmImageViewer.btnAutoClick(Sender: TObject);
begin
  BoldImage.StretchMode := bsmStretchProportional;
  BoldImage.AutoSize := False;
  BoldImage.Align := alClient;
end;

procedure TfrmImageViewer.btnScaleClick(Sender: TObject);
begin
  BoldImage.StretchMode := bsmStretchToScale;
  BoldImage.AutoSize := True;
  BoldImage.Align := alNone;
end;

procedure TfrmImageViewer.BoldImageResize(Sender: TObject);
begin
  BoldImage.Repaint;
  cboScale.Text := IntToStr(BoldImage.Scale);
end;

end.

File: examples/CBuilder/Simple/GUIControls/BoldImage/fViewAutoSize.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "fViewAutoSize.h"
//---------------------------------------------------------------------------
#pragma link "BoldHandles"
#pragma link "BoldRootedHandles"
#pragma link "BoldImage"
#pragma link "BoldReferenceHandle"
#pragma link "BoldExpressionHandle"
#pragma link "BoldSubscription"

#pragma resource "*.dfm"
TfrmImageViewer *frmImageViewer;
//---------------------------------------------------------------------------
__fastcall TfrmImageViewer::TfrmImageViewer(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmImageViewer::btnScale100Click(TObject *Sender)
{
  btnScale->Click();
  btnScale->Down = true;
  BoldImage->Scale = 100;
}
//---------------------------------------------------------------------------
void __fastcall TfrmImageViewer::cboScaleChange(TObject *Sender)
{
  btnScale->Click();
  btnScale->Down = true;
  BoldImage->Scale = StrToInt(cboScale->Text);
}
//---------------------------------------------------------------------------
void __fastcall TfrmImageViewer::btnAutoClick(TObject *Sender)
{
  BoldImage->StretchMode = bsmStretchProportional;
  BoldImage->AutoSize = false;
  BoldImage->Align = alClient;
}
//---------------------------------------------------------------------------
void __fastcall TfrmImageViewer::btnScaleClick(TObject *Sender)
{
  BoldImage->StretchMode = bsmStretchToScale;
  BoldImage->AutoSize = true;
  BoldImage->Align = alNone;
}
//---------------------------------------------------------------------------
void __fastcall TfrmImageViewer::BoldImageResize(TObject *Sender)
{
  BoldImage->Repaint();
  cboScale->Text = IntToStr(BoldImage->Scale);
}
//---------------------------------------------------------------------------
